600.java

```diff
@@ -65,29 +75,30 @@ public static void main(String[] args){
 	for(int i = 0; i < n; i++){
 		nums[i] = Integer.parseInt(strs[i]);
 	}
-	int[] order = new int[n];
+	int[] order = new int[n + 1];
 	for(int i = 0; i < n; i++){
```

Comment:
Ok, I am going to be a jerk here. I don't like order array because my head memory is so limited, and I can't remember those variable names. And I am a retard in programming, can you modify your code so there's no order array? Just nums and deque?